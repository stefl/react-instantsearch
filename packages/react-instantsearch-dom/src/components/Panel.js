import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import createClassNames from '../core/createClassNames';

const cx = createClassNames('Panel');

class Panel extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    header: PropTypes.node,
    footer: PropTypes.node,
  };

  static childContextTypes = {
    setCanRefine: PropTypes.func.isRequired,
  };

  static defaultProps = {
    className: '',
    header: null,
    footer: null,
  };

  state = {
    canRefine: true,
  };

  getChildContext() {
    return {
      setCanRefine: this.setCanRefine,
    };
  }

  setCanRefine = nextCanRefine => {
    this.setState({ canRefine: nextCanRefine });
  };

  render() {
    const { children, className, header, footer } = this.props;
    const { canRefine } = this.state;

    return (
      <div
        className={classNames(cx('', !canRefine && '-noRefinement'), className)}
      >
        {header && <div className={cx('header')}>{header}</div>}

        <div className={cx('body')}>{children}</div>

        {footer && <div className={cx('footer')}>{footer}</div>}
      </div>
    );
  }
}

export default Panel;
