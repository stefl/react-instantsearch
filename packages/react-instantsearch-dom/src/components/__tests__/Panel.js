import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Panel from '../Panel';

Enzyme.configure({ adapter: new Adapter() });

describe('Panel', () => {
  it('expect to render', () => {
    const wrapper = shallow(
      <Panel>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with custom className', () => {
    const props = {
      className: 'ais-Panel-Breadcrumb',
    };

    const wrapper = shallow(
      <Panel {...props}>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render without refinement', () => {
    const wrapper = shallow(
      <Panel>
        <div>Hello content</div>
      </Panel>
    );

    wrapper.setState({ canRefine: false });

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with header', () => {
    const props = {
      header: <p>Header</p>,
    };

    const wrapper = shallow(
      <Panel {...props}>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with footer', () => {
    const props = {
      footer: <p>Footer</p>,
    };

    const wrapper = shallow(
      <Panel {...props}>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to expose context to his children', () => {
    const wrapper = shallow(
      <Panel>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper.instance().getChildContext()).toEqual({
      setCanRefine: wrapper.instance().setCanRefine,
    });
  });

  it('expect to render when setCanRefine is called', () => {
    const wrapper = shallow(
      <Panel>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper.state('canRefine')).toBe(true);
    expect(wrapper).toMatchSnapshot();

    // Simulate context call
    wrapper
      .instance()
      .getChildContext()
      .setCanRefine(false);

    wrapper.update();

    expect(wrapper.state('canRefine')).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });
});
