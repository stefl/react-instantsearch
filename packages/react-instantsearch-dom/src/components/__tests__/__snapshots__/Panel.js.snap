// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel expect to render 1`] = `
<div
  className="ais-Panel"
>
  <div
    className="ais-Panel-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`Panel expect to render when setCanRefine is called 1`] = `
<div
  className="ais-Panel"
>
  <div
    className="ais-Panel-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`Panel expect to render when setCanRefine is called 2`] = `
<div
  className="ais-Panel ais-Panel--noRefinement"
>
  <div
    className="ais-Panel-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`Panel expect to render with custom className 1`] = `
<div
  className="ais-Panel ais-Panel-Breadcrumb"
>
  <div
    className="ais-Panel-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`Panel expect to render with footer 1`] = `
<div
  className="ais-Panel"
>
  <div
    className="ais-Panel-body"
  >
    <div>
      Hello content
    </div>
  </div>
  <div
    className="ais-Panel-footer"
  >
    <p>
      Footer
    </p>
  </div>
</div>
`;

exports[`Panel expect to render with header 1`] = `
<div
  className="ais-Panel"
>
  <div
    className="ais-Panel-header"
  >
    <p>
      Header
    </p>
  </div>
  <div
    className="ais-Panel-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`Panel expect to render without refinement 1`] = `
<div
  className="ais-Panel ais-Panel--noRefinement"
>
  <div
    className="ais-Panel-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;
